service/history/handler.go

```diff
@@ -163,19 +177,33 @@ func (h *Handler) Start() error {
 		}
 	}
 
+	if h.GetClusterMetadata().GetReplicationConsumerConfig().Type == config.ReplicationConsumerTypeRPC {
+		var replicationTaskFetchers []*replicationTaskFetcher
+		for sourceCluster, sourceFrontend := range h.Service.GetClientBean().GetRemoteFrontendClients() {
```

Comment:
```suggestion
		```
	numFetchers := h.GetClusterMetadata().GetReplicationConsumerConfig().NumFetchers
	for clusterName, info := range shard.GetService().GetClusterMetadata().GetAllClusterInfo() {
		if !info.Enabled {
			continue
		}

		if clusterName != shard.GetService().GetClusterMetadata().GetCurrentClusterName() {
			remoteFrontendClient := h.Service.GetClientBean().GetRemoteFrontendClient(clusterName)
			fetcher := newReplicationTaskFetcher(h.GetLogger(), clusterName, numFetchers, remoteFrontendClient)
			h.replicationTaskFetchers = append(h.replicationTaskFetchers, fetcher)
		}
	}

	for _, fetchers := range h.replicationTaskFetchers {
		fetchers.Start()
	}
```
```